src/plugins/query_enhancements/server/search/promql_search_strategy.ts

```diff
@@ -80,3 +81,49 @@ export const promqlSearchStrategyProvider = (
     },
   };
 };
+
+function createDataFrame(rawResponse: PrometheusResponse) {
+  const { result: series } = JSON.parse(rawResponse.result) as PrometheusResult;
```

Comment:
i thought opensearch API shouldn't return a JSON string? it should be an already parsed object

Reply:
I may need to pull the latest changes in the sql plugin. Currently the `results` I'm getting from the opensearch api is a json string.

Reply:
Would be great to avoid parsing this on the frontend

Reply:
i see in the code currently it indeed returns query results as a string. @goyamegh is it intentional or just temporary code to make it work?